eZ/Publish/Core/FieldType/Image/ImageStorage/Gateway/LegacyStorage.php

```diff
@@ -231,6 +231,29 @@ public function countImageReferences($uri)
         return (int)$statement->fetchColumn();
     }
 
+    public function isImageReferenced(string $uri): bool
+    {
+        $path = $this->redecorator->redecorateFromSource($uri);
+
+        $connection = $this->getConnection();
+
+        $selectQuery = $connection->createSelectQuery();
+        $selectQuery->select(1)
+        ->from(
+            $connection->quoteTable('ezimagefile')
+        )->where(
+            $selectQuery->expr->eq(
+                $connection->quoteColumn('filepath'),
+                $selectQuery->bindValue($path)
+            )
+        );
```

Comment:
Indentation needs adjustment.

```suggestion
            ->from(
                $connection->quoteTable('ezimagefile')
            )->where(
                $selectQuery->expr->eq(
                    $connection->quoteColumn('filepath'),
                    $selectQuery->bindValue($path)
                )
            );
```